configs/ruby/GPU_VIPER.py

```diff
@@ -1124,3 +1124,28 @@ def create_system(
     ruby_system.network.number_of_virtual_networks = 11
 
     return (cpu_sequencers, dir_cntrl_nodes, mainCluster)
+
+
+def RP_choose(test_name):
```

Comment:
This is a hardcoded version of gem5/configs/ObjectList.py::rp_list

Reply:
> This is a hardcoded version of gem5/configs/ObjectList.py::rp_list

Hi, from what my understanding now, the change does not work since, for example, if we want to make self.replacement_policy = TreePLRURP(), however the self.replacement_policy = ObjectList.rp_list.get(options.tcp_rp) will not make the same change to call the class TreePLRURP(), since it would generate an error that "could not create sytem for ruby protocol GPU_VIPER".

Reply:
I may not have understood your question. Are you saying that you wanted to use ObjectList.rp_list.get() to replace the assignments ´self.replacement_policy = TreePLRURP()´? If so, then you would probably have to do it like ´ObjectList.rp_list.get("TreePLRURP")´, IIRC, but I have never tried doing that. Regardless, this line is not the major focus of my suggestion for deduplication, so it does not need to be changed to use the object list. Are you having trouble making the line ´self.replacement_policy = ObjectList.rp_list.get(options.tcp_rp)´ work?

Reply:
Yes, I am not able to make the line  ´self.replacement_policy = ObjectList.rp_list.get(options.tcp_rp)´  work now, and that TreePLRURP is an example I had

Reply:
Could you give me a command line to run a test case so that I can try creating the PR?

Reply:
Sorry for the late response. You could run an example command line to both use RRIP for TCP and TCC:

docker run --volume $(pwd):$(pwd) -w $(pwd) docker.io/library/gccimage build/VEGA_X86/gem5.opt  configs/example/apu_se.py -n 3 --dgpu --gfx-version=gfx900 --num-compute-units=4 --cu-per-sa=4 --num-gpu-complex=1 --reg-alloc-policy=dynamic --num-tccs=8 --num-dirs=64 --mem-size=16GB --mem-type=HBM_2000_4H_1x64 --vreg-file-size=16384 --sreg-file-size=800 --tcc-size=4MB --gpu-clock=1801MHz --ruby-clock=1000MHz --vrf_lm_bus_latency=6 --mem-req-latency=69 --mem-resp-latency=69 --mandatory_queue_latency=1 --max-cu-tokens=160 --max-coalesces-per-cycle=10 --sqc-size=16kB --tcp-size=4MB --scalar-mem-req-latency=28 --TCP_latency=4 --tcp-assoc=16 --tcp-num-banks=16 --TCC_latency=121 --tcc-assoc=16 --tcc-tag-access-latency=1 --tcc-data-access-latency=1 --tcp-rp=RRIPRP --WB_L2 --tcc-rp=RRIPRP -c TESTNAME

The docker.io/library/gccimage can be built by 
cd util/dockerfiles/gcn-gpu/
docker build -t gccimage .

Reply:
Thanks @odanrc !  I'm not sure which docker @yuxiaojia is using, but in theory you should be able to test yours using the same instructions here: https://github.com/gem5/gem5-resources/blob/develop/src/gpu/square/README.md